crates/dslab-mp/src/mc/node.rs

```diff
@@ -9,15 +11,37 @@ use crate::message::Message;
 use crate::node::{EventLogEntry, ProcessEntry, ProcessEvent, TimerBehavior};
 use crate::process::ProcessState;
 
+#[derive(Clone)]
 pub struct ProcessEntryState {
-    pub proc_state: Box<dyn ProcessState>,
+    pub proc_state: Rc<dyn ProcessState>,
     pub event_log: Vec<EventLogEntry>,
     pub local_outbox: Vec<Message>,
     pub pending_timers: HashMap<String, u64>,
     pub sent_message_count: u64,
     pub received_message_count: u64,
 }
 
+impl Hash for ProcessEntryState {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.proc_state.hash().hash(state);
```

Comment:
Предлагаю изменить сигнатуру hash() у ProcessState, чтобы избежать создания отдельного hasher-а:

```
pub trait ProcessState: Downcast + Debug {
    fn hash(&self, hasher: &mut dyn Hasher);
}
```

тогда здесь будет

```suggestion
        self.proc_state.hash(state);
```

Reply:
Да, наверное смысл есть,  вроде так просто исторически было у Антона

Reply:
Вполне можно поменять

Reply:
поправил

Reply:
смог сделать полноценный трейт